test/distributed/_shard/sharded_tensor/test_sharded_tensor.py

```diff
@@ -1464,6 +1465,61 @@ def test_gather_uneven(self) -> None:
         else:
             self.assertIsNone(full_tensor)
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_to(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+        h, w = 10, 20
+        st = sharded_tensor.zeros(spec, h, w)
+
+        # Test ability to move st to CPU
+        # first test error if not passing pg
+        with self.assertRaisesRegex(AssertionError, 'pass in process_group'):
+            st.to(device="cpu")
+
+        gloo_pg = dist.new_group(backend="gloo")
+        st.to(device="cpu", process_group=gloo_pg)
+        spec = st.sharding_spec()
+        metas = st.metadata().shards_metadata
+        self.assertIsInstance(spec, ChunkShardingSpec)
```

Comment:
Can we validate that the spec is the same as passed in during tensor creation except that the device is now cpu?